### DISCUSSION TITLE
Developing research software: Choose "their" adventure!

### DISCUSSION THREAD

**Discussion by pr4deepr** [RSEAA/RSEAA.github.io#59] [Discussion topics for #RSEAA2023]
From my experience, researchers identify gaps in tools during their research, create something that works for them, and also realize there is value for the broader community. Some develop them further, but most of the time software engineering skills are severely lacking. It is ideal to engage software engineers from the start, but thats not always possible. 

Considering there may be the typical constraints, i.e. Access to budget, RSEs and compute.

How do you recommend they progress? What support is there?

 They can learn software engineering of course, but learning in tandem with an RSE maybe more efficient.

**Comment by tom-h** [comment]
There are lots of attempts to crystallise good practices that are often developed and published within research disciplines where software production is common. I think these are a great start, but they often focus on technical aspects.

What I haven't seen a lot of if guidance for identifying, targeting and building a community around software. For this, I think there is a fair bit of material out there in the community open source community.
